### PR TITLE
Feature: Implement StructMemberExpr LValue Generation

### LINKED ISSUE #438: Implement StructMemberExpr LValue Generation in GILGenLValue
## Description

The function `visitStructMemberExpr` in `GILGenLValue` currently contains a TODO and is not implemented:

```cpp
    gil::Value visitStructMemberExpr(StructMemberExpr *expr)
    {
        // TODO: implement this funciton
        return gil::Value::getEmptyKey();
    }
```

### Task
Implement the logic for generating a GIL lvalue for struct member access expressions (`StructMemberExpr`). This should:
- Properly handle struct member accesses as lvalues in the GIL generation phase.
- Ensure that the member exists and is accessible.
- Return the correct `gil::Value` representing the lvalue for the struct member.
- Add appropriate error handling and assertions as needed.

### Context
- The code is located in `lib/GILGen/GILGenLValue.hpp`.
- The relevant AST node is `StructMemberExpr` (see `include/AST/Expr/StructMemberExpr.hpp`).
- The function is part of the `GILGenLValue` struct, which is responsible for generating lvalues from AST nodes.

### References
- [StructMemberExpr definition](include/AST/Expr/StructMemberExpr.hpp)
- [GILGenLValue implementation](lib/GILGen/GILGenLValue.hpp)

### Acceptance Criteria
- The function is implemented and passes all relevant unit tests.
- The implementation is documented and follows project coding standards.
- Any new errors or edge cases are handled appropriately.

### PR DESCRIPTION
**Pull request by LindonAliu**
Closes #438

### CODE REVIEW

File: include/GIL/Instructions/StructFieldPtrInst.hpp [hunk 1]
```diff
@@ -7,6 +7,10 @@
 
 namespace glu::gil {
 
+// Target architecture constants for 64-bit systems
+constexpr size_t TARGET_POINTER_SIZE = 8;
+constexpr size_t TARGET_POINTER_ALIGNMENT = 8;
```
Comment (Snowy1803) [inline include/GIL/Instructions/StructFieldPtrInst.hpp:12]: the lib/GIL wouldn't have access to the information about the target — this info should be in the lib/GILGen

**Review by Snowy1803 — APPROVED** [review]
lgtm
